sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py

```diff
@@ -186,19 +167,33 @@ def _on_outcome(self, outcome, condition):
         self._outcome = outcome
         self._condition = condition
 
+    def _send_event_data(
+        self,
+        timeout_time: Optional[float] = None,
+        last_exception: Optional[Exception] = None,
+    ) -> None:
+        if self._unsent_events:
+            self._amqp_transport.send_messages(
+                self, timeout_time, last_exception, _LOGGER
+            )
+
+    def _send_event_data_with_retry(self, timeout: Optional[float] = None) -> None:
+        return self._do_retryable_operation(self._send_event_data, timeout=timeout)
```

Comment:
Later we might need to look at moving the "do_retryable_operation" into the transport as well - as this is built into pyamqp and so wont be needed later. (Don't need to change it now though).

Reply:
sounds good, created an issue for this here: #25700